sci-libs/fftw/fftw-3.3.8.recipe

```diff
@@ -63,33 +95,86 @@ BUILD_PREREQUIRES="
 	"
 
 defineDebugInfoPackage fftw$secondaryArchSuffix \
-	$libDir/libfftw3.so.$libVersion \
-	$libDir/libfftw3f.so.$libVersion
+	"$libDir"/libfftw3.so.$libVersion \
+	"$libDir"/libfftw3_threads.so.$libVersion \
+	"$libDir"/libfftw3f.so.$libVersion \
+	"$libDir"/libfftw3f_threads.so.$libVersion \
+	"$libDir"/libfftw3l.so.$libVersion \
+	"$libDir"/libfftw3l_threads.so.$libVersion \
+	"$libDir"/libfftw3q.so.$libVersion \
+	"$libDir"/libfftw3q_threads.so.$libVersion
 
 BUILD()
 {
-	make clean && make distclean || true
+	mkdir -p double long-double quad float
+
+	CONFIGURE="--enable-shared --enable-threads"
+if [ "$effectiveTargetArchitecture" != x86_gcc2 ]; then
+	CONFIGURE_X86="--enable-sse2 --enable-avx --enable-avx2 --enable-avx512"
+fi
 
 	# build double-precision version
-	runConfigure ./configure --enable-shared
+	cd double
+	runConfigure ../configure \
+		$CONFIGURE \
+		$CONFIGURE_X86
+	make $jobArgs
+	cd ..
+
+	# build long-double-precision version
+	cd long-double
+	runConfigure ../configure \
+		--enable-long-double \
+		$CONFIGURE
 	make $jobArgs
-	make install
+	cd ..
 
-	make clean && make distclean || true
+if [ "$effectiveTargetArchitecture" != x86_gcc2 ]; then
+	# build long-quad-precision version
+	cd quad
+	runConfigure ../configure \
+		--enable-quad-precision \
+		$CONFIGURE
+	make $jobArgs
+	cd ..
+fi
 
 	# build float-precision version
-	runConfigure ./configure --enable-float --enable-shared
+	cd float
+	runConfigure ../configure \
+		--enable-float \
+		$CONFIGURE \
+		$CONFIGURE_X86
 	make $jobArgs
-	make install
+	cd ..
 }
 
 INSTALL()
 {
+	if [ "$effectiveTargetArchitecture" != x86_gcc2 ]; then
+		for f in double long-double quad float; do
+	else
+		for f in double long-double float; do
+	fi
```

Comment:
```
INSTALL()
{
	if [ "$effectiveTargetArchitecture" != x86_gcc2 ]; then
		for f in double long-double quad float; do
	else
		for f in double long-double float; do
	fi
			cd $f
			make install
			cd ..
		done
```
This won't work. Unlike a `#if` + `#else` + `#endif`, the `if` + `else` + `fi`  can't include the `for` + `do` and let the `done` outside.

Reply:
The easiest way to address my previous comment is:
```diff
 INSTALL()
 {
 	if [ "$effectiveTargetArchitecture" != x86_gcc2 ]; then
-		for f in double long-double quad float; do
+		DIRS="double long-double quad float"
 	else
-		for f in double long-double float; do
+		DIRS="double long-double float"
 	fi
+	for f in $DIRS; do
- 		cd $f
 		make -C $f install
- 		cd ..
 	done
 
- 	rm $libDir/libfftw*.la
+ 	rm -f "$libDir"/libfftw*.la 
```
(The very last change, for the `rm $libDir/libfftw*.la` is not important.)

And of course, the `make -C $f install` saves us the `cd $f` and the `cd ..` so I think it can be a nice idea.

Reply:
Hmm, here's a more compact variant:
```diff
 INSTALL()
 {
 	if [ "$effectiveTargetArchitecture" != x86_gcc2 ]; then
-		for f in double long-double quad float; do
+		for f in double long-double quad float; do make -C $f install; done
 	else
-		for f in double long-double float; do
+		for f in double long-double float; do make -C $f install; done
 	fi
- 		cd $f
- 		make install
- 		cd ..
- 	done
```